### PR TITLE
Improved output viewer

### LINKED ISSUE #7959: Options to show/hide the comments and R-commands in output window no longer work?
@ChrisMarsh82 and @Patowhiz we used to have the option in Tools > Options > Output Window to show or hide the R-commands and the comments.
They no longer appear to work.
I wondered whether this might be related to Patrick's changes for html, but they don't work either in Version 0.7.7?
I would be happy for the options to be implemented differently, if that is your preference, but would like the option to not show R commands in the next release, if at all possible.

### PR DESCRIPTION
**Pull request by Patowhiz**
Fixes some items in issue #7852.
Fixes #7959.
It also adapts 3 variable frequencies dialog to use the new output system, effectively adding support for raw "grob" objects.
@africanmathsinitiative/developers  this is now ready for review.

### CODE REVIEW

**Review by rdstern — COMMENTED** [review]
@Patowhiz excellent.  I checked the 3-way frequencies and that seems to work.
And the hiding of commands also seems (almost) to work.  The one problem seems to be the comments.  It says initially the comments are on and the commands are off - perfect.
But the comments are not on!  They come on when I ask for the commands in the output window.  But they seem always to be off, if the commands are off.

Delighted that is almost working already!

**Comment by Patowhiz** [comment]
@rdstern there is an oddity with those 2 options. 
The current output structures don't separate comments from scripts. Comments are assumed to be part of the scripts.
To achieve what the 2 options were designed to do, we will have to separate them out during display.
That's doable but was lower in my priority list. Alternatively for the next release we could disable that option and assume it's part of the commands option.

Is it significant to you to have those comments in the outputs? I'm now making further enhancements to the output system,  including improving the structures as discussed by @ChrisMarsh82, I could look for a work around.

**Comment by rdstern** [comment]
@Patowhiz and @ChrisMarsh82 my memory is that display of comments and R-code used to be separate, but might now be joined in the new output window?  I am afraid Danny agrees that this would be good to separate these, before changing the default to not show the commands.  Perhaps @ChrisMarsh82 may also be able to see what's happening here?

**Comment by Patowhiz** [comment]
@rdstern I have now added the enabling and disabling of comments separately as suggested.
Notice how the comments becomes odd all over a sudden. It states "code generated". 
output

**Review by rdstern — COMMENTED** [review]
@Patowhiz this is great and just what we want.
I am happy to consider the most appropriate default comment to include.  I have changed it slightly to read `R code`, rather than Code and have a colon at the end.

Not the user can change the default comment - and we can continue to discuss the initial default:

image

The one small item to change is that when commands are off, we don't even need them at startup:

image

And it is perhaps quite nice to have something there.  So we can leave the top comment.  Then I am relaxed about whether we show the other 2 comments or not.  If we do, then I have a minor point about the stars one, because (correctly) it is switched off.  But the comment looks as though they will be shown.  We could omit that comment, or perhaps add: Off if the option remains unchecked, and On if it is checked.  But don't spend more than 5 minutes on that feature.  If it will take time, then just don't show that comment.

But let me emphasise the importance of the comments.  If we can think of a friendlier default then I am happy.  But the idea is that the comments field is on every dialogue!  And, just like promoting the addition of comments to programmers in their code I would like to promote that users include useful comments about each bit of output.  Perhaps why they are doing it, or what they expect, etc.  So that is only the comment if users are too lazy, or busy to add a better one!

**Review by rdstern — COMMENTED** [review]
@Patowhiz I see the reason you did that, but I was assuming you would keep the comment  at the top about the working directory and then not show either that bit of R or anything else below that - except (possibly) for an editable welcoming comment, e.g. "Welcome to R-Instat".

**Comment by Patowhiz** [comment]
Did what exactly?

**Comment by Patowhiz** [comment]
@rdstern @lloyddewit this is now ready for review.
@rdstern like I said before, when you switch off the commands, the default dialog comments don't make sense. I suggest changing the default comments to something that mentions "output generated by..." instead of "code generated by...". 
That can be done in a separate.

**Comment by Patowhiz** [comment]
I've noticed the "Last Graph" menu item does not send the code to the output window. I'll add that back in a new PR that relates to using objects. Thanks.

**Review by rdstern — APPROVED** [review]
@Patowhiz the graphs also appear full size.  So I am happy and would like this to be merged as soon as possible.  Then we can play and critique.  Can @ChrisMarsh82 or @lloyddewit check the code and merge quickly, or is that a major job?

**Review by lloyddewit — COMMENTED** [review]
@Patowhiz It looks like you simplified and cleaned the code. The new code is clear and readable - thank you.
I only had one open question. If you can check this, then we can merge

File: instat/UserControl/ucrOutputPage.vb [hunk 1]
```diff
@@ -128,20 +129,177 @@ Public Class ucrOutputPage
...
+         .BorderStyle = BorderStyle.None
+        }
+
+        'if settings are not available or both show commands and comments settings are enabled then just show the whole script
+        If frmMain.clsInstatOptions Is Nothing OrElse (frmMain.clsInstatOptions.bIncludeCommentDefault AndAlso frmMain.clsInstatOptions.bCommandsinOutput) Then
+            FillRichTextBoxWithFormatedRScript(richTextBox, outputElement.FormattedRScript)
+        Else
+
+            'if either show commands or comments settings is enabled show the script that corresponds to either
+            If frmMain.clsInstatOptions.bIncludeCommentDefault Then
```
[the remark this answers is not in the document]
Comment (Patowhiz) [reply instat/UserControl/ucrOutputPage.vb:164]: It is show/display comments only. Those comments are selected from the script.
That section is only executed if either of the 2 options are enabled. Not when both are enabled.

[the remark this answers is not in the document]
Comment (lloyddewit) [reply instat/UserControl/ucrOutputPage.vb:164]: understood. Thanks for the explanation

**Review by lloyddewit — APPROVED** [review]

**Pull request merged**